standalone/src/test/java/io/pravega/local/PravegaEmulatorResource.java

```diff
@@ -95,6 +94,32 @@ public PravegaEmulatorResource(boolean authEnabled, boolean tlsEnabled, boolean
         pravega = emulatorBuilder.build();
     }
 
+    /*
+    Implemented Builder pattern in java to make way for optional parameters such as "tlsProtocolVersion"
+     */
+    public static class PravegaEmulatorResourceBuilder {
```

Comment:
You can achieve exactly the same thing by decorating the constructor you have just removed above with `@Builder`. Lombok will auto-generate this builder for you. 

Please check out other cases where we used the builder pattern (search for `@Builder` in the codebase). The builder is the one that creates the class, and the class does not have a constructor that takes in a builder!

Reply:
Fixed. Used @Builder annotation for implementing builder pattern.